src/Widgets/TerminalWidget.vala

```diff
@@ -460,11 +451,26 @@ namespace Terminal {
                 paste_action.set_enabled (can_paste);
             });
 
-            // Update the "Copy Last Outptut" menu option
+            // Update the "Copy Last Output" menu option
             var has_output = !resized && get_last_output ().length > 0;
             copy_output_action.set_enabled (has_output);
         }
 
+        private void popup_context_menu (Gdk.Rectangle rect) {
+            main_window.update_context_menu ();
+            setup_menu ();
+
+            // Popup context menu below cursor position
+            context_popover = new Gtk.PopoverMenu () {
+                relative_to = this,
+                pointing_to = rect
+            };
+
+            context_popover.bind_model (context_menu_model, null);
+            context_popover.show_all ();
+            context_popover.popup ();
```

Comment:
I think `show_all` and `popup` are redundant here. I'm not sure `show_all` is necessary when using a menu from a model

Reply:
It does not work in Gtk3 without these.  Revisit in Gtk4